shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/product_details/ProductDetailsViewModel.kt

```diff
@@ -47,14 +55,42 @@ class ProductDetailsViewModel(
         }
     }
 
-    fun onWantClicked() {
+    fun onWantClicked(
+        productDetailsOpenedFrom: ProductDetailsOpenedFrom,
+    ) {
         menuProductDetailsState.value.menuProduct?.let { menuProduct ->
+            sendOnWantedClickedAnalytic(
+                menuProductUuid = menuProduct.uuid,
+                productDetailsOpenedFrom = productDetailsOpenedFrom
+            )
             sharedScope.launch {
                 addCartProductUseCase(menuProduct.uuid)
             }
         }
     }
 
+    private fun sendOnWantedClickedAnalytic(
+        menuProductUuid: String,
+        productDetailsOpenedFrom: ProductDetailsOpenedFrom,
+    ) {
+        when (productDetailsOpenedFrom) {
```

Comment:
Let's make it shorter - add the `event` variable returned from `when` and then call `sendEvent` only once